### PR TITLE
bump istio api and client in go.mod to v1.15.2

### LINKED ISSUE #996: outdated istio api
I am currently using a DestinationRule with load balancer setting of `LEAST_REQUEST` which was introduced in Istio 1.14 ([changelog](https://istio.io/latest/news/releases/1.14.x/announcing-1.14/change-notes/))
```
apiVersion: networking.istio.io/v1beta1
kind: DestinationRule
metadata:
  name: demo-least-request
  namespace: istio-system
spec:
  host: '*.svc.cluster.local'
  trafficPolicy:
    loadBalancer:
      simple: LEAST_REQUEST       # enum not supported by net-istio
```
This is currently preventing reconciliation of ingress due to the following error:
```
failed to list *v1alpha3.DestinationRule: unknown value "LEAST_REQUEST" for enum istio.networking.v1alpha3.LoadBalancerSettings_SimpleLB
```

Turns out that the dependency for istio api is not up to date.

### PR DESCRIPTION
**Pull request by a7i**
# Changes

- 🐛  Fix bug when ingress is not reconciled when a DestinationRule with loadbalancer setting `LEAST_REQUEST` is defined (available in istio 1.14 and above)

/kind bug

Fixes #996 

**Release Note**

```release-note
bump istio api and client libraries to 1.15.2
```

**Docs**

```docs

```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -8,8 +8,8 @@ require (
 	github.com/hashicorp/golang-lru v0.5.4
 	go.uber.org/zap v1.19.1
 	golang.org/x/sync v0.0.0-20220722155255-886fb9371eb4
-	istio.io/api v0.0.0-20210713195055-3a340e4f154e
-	istio.io/client-go v1.10.3
+	istio.io/api v0.0.0-20221010183118-b3cf9e87a71b
+	istio.io/client-go v1.15.3-0.20221010183518-9b4f78954e90
```
Comment (dprotaso) [inline go.mod:12]: can we pin this to released versions - ie. v1.15.2

**Comment by a7i** [comment]
Pulled in changes for PR https://github.com/knative/pkg/pull/2613 to use the new flag

**Pull request closed without merging**